http-client/src/main/java/io/airlift/http/client/HttpUriBuilder.java

```diff
@@ -55,7 +57,7 @@ private HttpUriBuilder(URI previous)
         scheme = previous.getScheme();
         host = previous.getHost();
         port = previous.getPort();
-        path = percentDecode(previous.getRawPath());
+        path = isNull(previous.getRawPath()) ? null : percentDecode(previous.getRawPath());
```

Comment:
An alternative way to write this is

Optional.ofNullable(...).map(...).orElse(null)

This avoids repeating the input expression, though I'm not sure it's an improvement over traditional way.